api/unstable/async_offload.h

```diff
@@ -0,0 +1,83 @@
+/*
+* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#pragma once
+
+#include <s2n.h>
+
+/**
+ * @file async_offload.h
+ * 
+ * The following APIs enable applications to offload expensive handshake operations that do not require user input.
+ * Use s2n_config_set_async_offload_callback() to configure the async callback and the set of operations to offload.
+ * 
+ * `s2n_negotiate()` will throw an `S2N_ERR_T_BLOCKED` error if the handshake is blocked by the offloading callback.
+ * s2n_async_op_perform() must be invoked to unblock the handshake.
+ */
+
+/**
+ * Opaque struct for the async offloading operation
+ */
+struct s2n_async_op;
+
+/**
+ * The type of operations supported by the async offloading callback. Each type is represented by a different bit.
+ * 
+ * WARNING: S2N_ASYNC_ALLOW_ALL will automatically opt in to all the new types added in the future.
```

Comment:
Should this be a warning? Don't we expect / encourage customers to use S2N_ASYNC_ALLOW_ALL?